### DISCUSSION TITLE
Split Strings

### DISCUSSION THREAD

**Discussion by yAwdniK** [codewars/codewars.com#2773] [Q&A]
help me please with (return nil)
what im doing wrong?
it works on my IDE , but doesnt in codewars.
![image](https://user-images.githubusercontent.com/73609179/184653045-47184de6-d39a-465c-87e6-ea0fc89811ee.png)
![image](https://user-images.githubusercontent.com/73609179/184653124-11ad6445-ae9b-45dd-a245-667d1ddb5caf.png)
![image](https://user-images.githubusercontent.com/73609179/184663534-e844ee75-ae11-43b9-aa7a-be3c086a2aaa.png)

**Comment by moocstudent** [comment]
I am not sure,But according my codewars experience,The IDE can use external libraries and JDK method automatically,
But in the codewars online IDE,the libraries or JDK method must import manually.
Wish this can help you!

**Comment by yAwdniK** [comment]
I'm done with this answer. My difficult is 7-8 kyu, but I'm done this for
myself xd. Hard for me

ср, 24 авг. 2022 г., 16:31 moocstudent ***@***.***>:

**Comment by hathawayANdRX105** [comment]
![image](https://user-images.githubusercontent.com/61958173/188491528-d230204e-5593-459c-a11b-d99713d1a03f.png)
Problem：
I think the problem is that you want to split each tow letter into answer array, but you use teststring in code. 'teststring' just shouldn't be the correct split string.

Suggests:
And it's kind of redundant for combining "'' + answer[i] + "'," and then append answer[i] to sliceAnswer.The `answer` slice is the correct form after concatenated string.
Another idea is that turns `str` into []byte slice, iterates it each tow char, append `'` or `',` in middle of interatrion, because string is immutable byte and every concatenation will copy it again which costs some memory and wastes time.

My English is not very good, I hope you can understand what I mean ‘ V ’
